server/routes/datesRoutes.js

```diff
@@ -0,0 +1,36 @@
+const dateParser = require('../services/Parser')
+const parseIt = new dateParser()
+const express = require('express');
+const { all } = require('./mealtimesRoutes');
+const router = express.Router()
+
+
+router.get('/dates', (req, res) => {
+    let dateReturned
+    try{
+       const { date } = req.query
+       if(date){
+        dateReturned = parseIt.getDate(date)
+       }
+       const allDates = parseIt.getAllDates()
+       let data = !dateReturned ? allDates : dateReturned
+       
+        if(data.length < 1){
```

Comment:
I would argue for making this 
```suggestion
        if(data.length === 0)
```
Because it is specific as to what you're testing for